### ISSUE TITLE
Investigate what happens if application terminate while activity is popped from stack

### ISSUE THREAD

**Issue by sfmskywalker** [elsa-workflows/elsa-core#104]
Scenario:

- `WorkflowInvoker` popped an activity to be executed.
- During execution of activity (e.g. `OnResume`), application terminates.

Questions:

1. Will the workflow stay halted on the activity that was being executed when the app terminated?

The desired result is that the workflow will execute the activity again.
Todo: figure out at which point the blocking activity is removed from the workflow instance, and at which point the workflow instance is persisted.

Perhaps a solid implementation is to only update the workflow instance with the removed blocking activity after the activity.resume method has completed. If it doesn't complete due to app termination, then the blocking activities will still sit on the workflow instance, ready to be retried when the workflow is next invoked in a new app instance.

**Comment by sfmskywalker** [comment]
Looks like it's all good:

1. Before a workflow instance is resumed, a new `Workflow` instance is created, which receives a copy of all blocking activities
2. When the `WorkflowExecutionContext` is created, it removes blocking activities from the `Workflow` object, not the `WorkflowInstance` object.
3. When a workflow run is finalized (when there are no more scheduled activities to execute), the `Workflow` object is serialized to a `WorkflowInstance`, which is then persisted.

In conclusion, this means that if the app terminates during the resumption of an activity, the next time the workflow is resumed, it will resume that same activity.
